src/Ocelot/Middleware/OcelotMiddlewareExtensions.cs

```diff
@@ -155,7 +155,16 @@ private static bool AdministrationApiInUse(IAdministrationPath adminPath)
         }
 
         private static async Task SetFileConfigInConsul(IApplicationBuilder builder,
-            IFileConfigurationRepository fileConfigRepo, IOptions<FileConfiguration> fileConfig,
+          IFileConfigurationRepository fileConfigRepo, IOptionsMonitor<FileConfiguration> fileConfig,
+          IInternalConfigurationCreator internalConfigCreator, IInternalConfigurationRepository internalConfigRepo)
+        {
+            fileConfig.OnChange(async (config) => await SetFileConfigInConsul(builder, fileConfigRepo, fileConfig, internalConfigCreator, internalConfigRepo));
```

Comment:
Actually I thought about this more and we shouldnt do it, the whole point is the config is in consul and it doesnt actually write it to disk.